### ISSUE TITLE
Compiler aborts on unsigned (ui32/ui64) ElementsAttr iteration: getValues<int> called on tensor<ui32>

### ISSUE THREAD

**Issue by ctr-pmuruganTT** [tenstorrent/tt-mlir#9031]
## Summary
A pass/conversion in tt-mlir iterates an integer `ElementsAttr` with the signed C++ type `int` (`attr.getValues<int>()` / `getValues<int64_t>()`) without checking the attribute's element-type signedness. When the constant is **unsigned** (`ui32`/`ui64`), MLIR raises and the process aborts:

```
ElementsAttr does not provide iteration facilities for type `int`, see attribute: dense<4294967295> : tensor<ui32>
invalid `T` for ElementsAttr::getValues
Signal: Aborted (6)
```

`4294967295` = `2^32 - 1` = `-1` reinterpreted as `ui32` — the "no-index" sentinel produced by `max_pool2d_with_indices`.

## Impact
- Any graph containing an unsigned integer `ElementsAttr` fails to compile with a hard `abort()` (not a recoverable diagnostic).
- Surfaced via **tt-xla** on `tests/runner/test_models.py::test_all_models_torch[efficientdet/pytorch-D0-single_device-training]`: **passes at `cf9cfb79`, fails at `7b70f5f6`** (the tt-xla PyTorch 2.11 / vLLM 0.20.2 uplift, PR tenstorrent/tt-xla#5568). Under torch 2.11, `torch_xla` emits integer indices/constants as **unsigned** (`ui32`/`ui64`) where torch 2.10 emitted signed, so pre-existing graphs now hit this path. It is training-specific because `max_pool2d_with_indices` (indices needed for the backward pass) is what produces the `ui32` sentinel.

## Minimal reproducer (torch + tt backend)
```python
import torch, tt_torch  # registers the "tt" backend

def f(x):
    # indices path -> ui32 index tensor + dense<4294967295> : tensor<ui32> sentinel
    out, idx = torch.nn.functional.max_pool2d(x, 3, stride=2, padding=1, return_indices=True)
    return out, idx

x = torch.randn(1, 8, 32, 32)
torch.compile(f, backend="tt")(x)   # -> aborts in tt-mlir
```

The StableHLO handed to tt-mlir contains unsigned index tensors and the offending constant:
```mlir
%cst = stablehlo.constant dense<4294967295> : tensor<ui32>
...
stablehlo.return %a, %b : tensor<ui32>, tensor<2048xui32>
%r = stablehlo.reshape %idx : (tensor<2048xui32>) -> tensor<1x8x16x16xui32>
return %out, %r : tensor<1x8x16x16xf32>, tensor<1x8x16x16xui32>
```
(In efficientdet the same `dense<4294967295> : tensor<ui32>` also appears as a `stablehlo.reduce_window` init value for the max-pool.)

## Root cause
The iterating call selects the C++ element type by the *number of bits* / a fixed signed type, ignoring the attribute's `unsigned` semantics. `mlir::ElementsAttr::getValues<T>()` only supports `T` compatible with the attribute's element type; `getValues<int>()` on a `ui32` attribute is invalid and asserts.

## Suggested fix
In the offending pass, dispatch the iteration type on the element type's signedness/bitwidth, e.g.:
- `ui32` → `getValues<uint32_t>()`, `ui64` → `getValues<uint64_t>()`
- `i32` → `getValues<int32_t>()`, `i64` → `getValues<int64_t>()`

(or normalize `APInt` via `getValues<llvm::APInt>()` and read `getZExtValue()/getSExtValue()` per signedness). A grep for `getValues<int` / `getValues<int64_t` over the tt-mlir conversion passes should locate the site(s).

## Environment
- tt-xla `7b70f5f6` (PyTorch 2.11.0 + vLLM 0.20.2 uplift, #5568); tt-mlir pin `260d4c495`.
- torch 2.11.0, torch_xla 2.9.0+git56f8e61, n150 (wormhole_b0).
